nwg_panel/modules/sway_workspaces.py

```diff
@@ -201,25 +202,22 @@ def refresh(self):
                         self.layout_icon.hide()
 
     def update_icon(self, win_id, win_name):
+        loaded_icon = False
         if win_id and win_name:
-            icon_from_desktop = get_icon_name(win_id)
-            if icon_from_desktop:
-                if "/" not in icon_from_desktop and not icon_from_desktop.endswith(
-                        ".svg") and not icon_from_desktop.endswith(".png"):
-                    update_image(self.icon, icon_from_desktop, self.settings["image-size"], self.icons_path)
-                else:
-                    pixbuf = GdkPixbuf.Pixbuf.new_from_file_at_size(icon_from_desktop, self.settings["image-size"],
-                                                                    self.settings["image-size"])
-                    self.icon.set_from_pixbuf(pixbuf)
-            else:
-                image = Gtk.Image()
-                update_image(image, win_id, self.settings["image-size"], self.icons_path)
+            try:
+                update_image_fallback_desktop(self.icon,
+                                              win_id,
+                                              self.settings["image-size"],
+                                              self.icons_path,
+                                              fallback=False)
+                loaded_icon = True
```

Comment:
mark success. won't reach this line if `update_image_fallback_desktop` raises an exception